File: nodejs/1_text_to_sfx.ts
#!/usr/bin/env node
/**
 * Example 1: Generate SFX from Text Description Only
 *
 * This example demonstrates how to generate sound effects using only a text description,
 * without any video input. Perfect for creating audio assets from scratch.
 */

import fs from "fs";
import path from "path";
import fetch from "node-fetch";

// Configuration
const API_KEY = process.env.MIRELO_API_KEY;
const BASE_URL = "https://api.mirelo.ai";
const OUTPUT_DIR = path.join(process.cwd(), "output", "example1");

if (!API_KEY) {
  throw new Error("Please set MIRELO_API_KEY environment variable");
}

// Create output directory
if (!fs.existsSync(OUTPUT_DIR)) {
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });
}

async function generateTextToSfx() {
  console.log("🎵 Generating SFX from text description...");
  console.log("-".repeat(60));

  // Text description of the sound you want to generate
  const textPrompt = "thunder and lightning storm with heavy rain";

  // Request parameters
  const payload = {
    text_prompt: textPrompt,
    duration: 5, // 5 seconds
    num_samples: 2, // Generate 2 variations
    model_version: "1.5" as const, // Use latest model
    creativity_coef: 7, // Higher creativity (1-10)
    return_audio_only: true,
  };

  console.log(`📝 Text prompt: ${textPrompt}`);
  console.log(`⏱️  Duration: ${payload.duration} seconds`);
  console.log(`🎲 Samples: ${payload.num_samples}`);
  console.log(`🎨 Creativity: ${payload.creativity_coef}/10`);
  console.log();

  // Make API request
  const response = await fetch(`${BASE_URL}/video-to-sfx`, {
    method: "POST",
    headers: {
      "x-api-key": API_KEY,
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });

  if (response.status !== 201) {
    console.log(`❌ Error: ${response.status}`);
    console.log(await response.json());
    return;
  }

  const result = (await response.json()) as { output_paths: string[] };
  const audioUrls = result.output_paths || [];

  console.log(`✅ Generated ${audioUrls.length} audio variations`);
  console.log();

  // Download generated audio files
  for (let idx = 0; idx < audioUrls.length; idx++) {
    const audioUrl = audioUrls[idx];
    console.log(`📥 Downloading variation ${idx + 1}...`);

    const audioResponse = await fetch(audioUrl);
    if (audioResponse.ok) {
      const audioBuffer = await audioResponse.arrayBuffer();
      const outputPath = path.join(OUTPUT_DIR, `sfx_variation_${idx + 1}.wav`);
      fs.writeFileSync(outputPath, Buffer.from(audioBuffer));
      console.log(`   Saved to: ${outputPath}`);
    } else {
      console.log(`   ❌ Failed to download: ${audioResponse.status}`);
    }
  }

  console.log();
  console.log("=".repeat(60));
  console.log("✨ Done! Check the output directory for your audio files.");
  console.log(`   Location: ${path.resolve(OUTPUT_DIR)}`);
  console.log();
}

// Run the example
generateTextToSfx().catch((error) => {
  console.error("❌ Error:", error);
  process.exit(1);
});

File: python/1_text_to_sfx.py
#!/usr/bin/env python3
"""
Example 1: Generate SFX from Text Description Only

This example demonstrates how to generate sound effects using only a text description,
without any video input. Perfect for creating audio assets from scratch.
"""

import os
import requests
from pathlib import Path

# Configuration
API_KEY = os.getenv("MIRELO_API_KEY")
BASE_URL = "https://api.mirelo.ai"
OUTPUT_DIR = Path("output/example1")

if not API_KEY:
    raise ValueError("Please set MIRELO_API_KEY environment variable")

# Create output directory
OUTPUT_DIR.mkdir(parents=True, exist_ok=True)

def generate_text_to_sfx():
    """Generate SFX from text description without video"""
    
    print("🎵 Generating SFX from text description...")
    print("-" * 60)
    
    # Text description of the sound you want to generate
    text_prompt = "thunder and lightning storm with heavy rain"
    
    # Request parameters
    payload = {
        "text_prompt": text_prompt,
        "duration": 5,  # 5 seconds
        "num_samples": 2,  # Generate 2 variations
        "model_version": "1.5",  # Use latest model
        "creativity_coef": 7,  # Higher creativity (1-10)
        "return_audio_only": True,
    }
    
    print(f"📝 Text prompt: {text_prompt}")
    print(f"⏱️  Duration: {payload['duration']} seconds")
    print(f"🎲 Samples: {payload['num_samples']}")
    print(f"🎨 Creativity: {payload['creativity_coef']}/10")
    print()
    
    # Make API request
    response = requests.post(
        f"{BASE_URL}/video-to-sfx",
        headers={
            "x-api-key": API_KEY,
            "Content-Type": "application/json"
        },
        json=payload
    )
    
    if response.status_code != 201:
        print(f"❌ Error: {response.status_code}")
        print(response.json())
        return
    
    result = response.json()
    audio_urls = result.get("output_paths", [])
    
    print(f"✅ Generated {len(audio_urls)} audio variations")
    print()
    
    # Download generated audio files
    for idx, audio_url in enumerate(audio_urls, 1):
        print(f"📥 Downloading variation {idx}...")
        
        audio_response = requests.get(audio_url)
        if audio_response.status_code == 200:
            output_path = OUTPUT_DIR / f"sfx_variation_{idx}.wav"
            output_path.write_bytes(audio_response.content)
            print(f"   Saved to: {output_path}")
        else:
            print(f"   ❌ Failed to download: {audio_response.status_code}")
    
    print()
    print("=" * 60)
    print("✨ Done! Check the output directory for your audio files.")
    print(f"   Location: {OUTPUT_DIR.absolute()}")
    print()

if __name__ == "__main__":
    generate_text_to_sfx()

